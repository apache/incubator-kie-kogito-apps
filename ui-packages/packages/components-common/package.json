{
  "name": "@kogito-apps/components-common",
  "version": "1.0.0",
  "description": "This library provides a set of common React components for use with the Kogito Components.",
  "private": true,
  "main": "dist/index.js",
  "types": "dist/index.d.ts",
  "author": "",
  "license": "Apache-2.0",
  "repository": {
    "type": "git",
    "url": "https://github.com/kiegroup/kogito-apps.git"
  },
  "files": [
    "dist/"
  ],
  "scripts": {
    "precommit": "lint-staged",
    "copy:css": "copyfiles -u 1 \"src/**/*.{sass,scss,css,svg,png}\" dist/",
    "build": "rimraf dist && tsc -m commonjs -p tsconfig.json && yarn run copy:css",
    "build:prod": "yarn run clean && yarn run lint && yarn run build",
    "test:report": "yarn test --ci --reporters=jest-junit",
    "test": "jest --runInBand --ci --reporters=default --reporters=jest-junit",
    "test:coverage": "rimraf coverage && yarn test --coverage",
    "update-snapshot": "jest --updateSnapshot",
    "lint": "eslint './src/**/*.ts{,x}'",
    "format": "prettier --config ../../.prettierrc --check --write './src/**/*.{tsx,ts,js}'",
    "clean": "rimraf dist"
  },
  "dependencies": {
    "@kogito-apps/patternfly-base": "^1.0.0",
    "ajv": "^6.12.6",
    "uniforms": "^3.5.5",
    "uniforms-bridge-json-schema": "^3.5.5",
    "uniforms-patternfly": "^4.7.3"
  },
  "devDependencies": {
    "@kogito-apps/ouia-tools": "^1.0.0",
    "babel-loader": "8.1.0",
    "@types/json-schema": "7.0.7"
  },
  "jest": {
    "setupFiles": [
      "../../config/Jest-config/test-shim.js",
      "../../config/Jest-config/test-setup.js",
      "core-js"
    ],
    "moduleFileExtensions": [
      "ts",
      "tsx",
      "js"
    ],
    "coveragePathIgnorePatterns": [
      "./src/static",
      "./src/graphql",
      "dist/"
    ],
    "coverageReporters": [
      [
        "lcov",
        {
          "projectRoot": "../../"
        }
      ]
    ],
    "snapshotSerializers": [
      "enzyme-to-json/serializer"
    ],
    "transformIgnorePatterns": [],
    "transform": {
      "^.+\\.jsx?$": "../../config/Jest-config/babel-jest-wrapper.js",
      "^.+\\.(ts|tsx)$": "ts-jest",
      "\\.(jpg|jpeg|png|svg)$": "../../config/Jest-config/fileMocks.js"
    },
    "testMatch": [
      "**/tests/*.(ts|tsx|js)"
    ],
    "moduleNameMapper": {
      "\\.(scss|sass|css)$": "identity-obj-proxy"
    }
  },
  "lint-staged": {
    "*.{ts,tsx}": [
      "yarn run format",
      "yarn run lint"
    ]
  }
}