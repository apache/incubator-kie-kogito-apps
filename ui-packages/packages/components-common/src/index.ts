/*
 * Copyright 2021 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

export {
  default as DataTable,
  DataTableColumn
} from './components/DataTable/DataTable';
export {
  default as ItemDescriptor,
  ItemDescription
} from './components/ItemDescriptor/ItemDescriptor';
export {
  default as KogitoEmptyState,
  KogitoEmptyStateType
} from './components/KogitoEmptyState/KogitoEmptyState';
export { default as KogitoSpinner } from './components/KogitoSpinner/KogitoSpinner';
export { default as LoadMore } from './components/LoadMore/LoadMore';
export { default as ServerErrors } from './components/ServerErrors/ServerErrors';
export { default as EndpointLink } from './components/EndpointLink/EndpointLink';
export { default as FormRenderer } from './components/FormRenderer/FormRenderer';
export {
  default as FormNotification,
  Notification
} from './components/FormNotification/FormNotification';
export { default as FormFooter } from './components/FormFooter/FormFooter';
export * from './components/utils/FormActionsUtils';
export * from './components/utils/FormValidator';
export * from './components/utils/ModelConversionTool';
export * from './types';
