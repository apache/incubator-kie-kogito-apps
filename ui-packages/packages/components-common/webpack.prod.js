/*
 * Copyright 2021 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

const path = require('path');
const merge = require('webpack-merge');
const common = require('./webpack.common.js');
const MiniCssExtractPlugin = require('mini-css-extract-plugin');
const OptimizeCSSAssetsPlugin = require('optimize-css-assets-webpack-plugin');

module.exports = merge(common, {
  mode: 'production',
  devtool: 'source-map',
  optimization: {
    minimizer: [new OptimizeCSSAssetsPlugin({})]
  },
  plugins: [
    new MiniCssExtractPlugin({
      filename: 'src/components/styles.css',
      chunkFilename: '[name].bundle.css'
    })
  ],
  module: {
    rules: [
      {
        test: /\.css$/,
        include: [
          path.resolve(__dirname, 'src'),
          path.resolve('../../node_modules/patternfly'),
          path.resolve('../../node_modules/@patternfly/patternfly'),
          path.resolve('../../node_modules/@patternfly/react-styles/css'),
          path.resolve(
            '../../node_modules/@patternfly/react-core/dist/styles/base.css'
          ),
          path.resolve(
            '../../node_modules/@patternfly/react-core/dist/esm/@patternfly/patternfly'
          ),
          path.resolve(
            '../../node_modules/@patternfly/react-core/node_modules/@patternfly/react-styles/css'
          ),
          path.resolve(
            '../../node_modules/@patternfly/react-table/node_modules/@patternfly/react-styles/css'
          ),
          path.resolve(
            '../../node_modules/react-calendar/dist/Calendar.css'
          ),
          path.resolve(
            '../../node_modules/react-clock/dist/Clock.css'
          ),
          path.resolve(
            '../../node_modules/react-datetime-picker/dist/DateTimePicker.css'
          )
        ],
        use: [{
          loader: MiniCssExtractPlugin.loader,
          options: {
            publicPath: '../../',
          }
        }, 'css-loader']
      }
    ]
  }
});