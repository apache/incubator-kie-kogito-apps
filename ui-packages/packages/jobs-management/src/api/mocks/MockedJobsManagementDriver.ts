import { JobsManagementDriver } from '../JobsManagementDriver';

export const MockedJobsManagementDriver = jest.fn<JobsManagementDriver, []>(
  () => ({
    initialLoad: jest.fn(),
    applyFilter: jest.fn(),
    bulkCancel: jest.fn(),
    cancelJob: jest.fn(),
    rescheduleJob: jest.fn(),
    sortBy: jest.fn(),
    query: jest.fn()
  })
);
