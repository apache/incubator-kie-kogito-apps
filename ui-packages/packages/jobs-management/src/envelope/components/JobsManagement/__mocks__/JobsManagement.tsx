import React from 'react';

const MockedJobsManagement = (): React.ReactElement => {
  return <></>;
};

export default MockedJobsManagement;
