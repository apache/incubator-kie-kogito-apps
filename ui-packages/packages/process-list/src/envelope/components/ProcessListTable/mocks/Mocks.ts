/*
 * Copyright 2021 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import {
  ProcessInstance,
  ProcessInstanceState
} from '@kogito-apps/management-console-shared';
export const ProcessInstances: ProcessInstance[] = [
  {
    id: '538f9feb-5a14-4096-b791-2055b38da7c6',
    processId: 'travels',
    businessKey: 'Tra234',
    parentProcessInstanceId: null,
    parentProcessInstance: null,
    processName: 'travels',
    rootProcessInstanceId: null,
    roles: [],
    state: ProcessInstanceState.Error,
    addons: ['jobs-management', 'prometheus-monitoring', 'process-management'],
    start: new Date('2019-10-22T03:40:44.089Z'),
    error: {
      nodeDefinitionId: '__a1e139d5-4e77-48c9-84ae-34578e9817n',
      message: 'Something went wrong'
    },
    lastUpdate: new Date('2019-10-22T03:40:44.089Z'),
    serviceUrl: 'http://localhost:4000',
    endpoint: 'http://localhost:4000',
    variables:
      '{"flight":{"arrival":"2019-10-30T22:00:00Z[UTC]","departure":"2019-10-23T22:00:00Z[UTC]","flightNumber":"MX555"},"trip":{"begin":"2019-10-23T22:00:00Z[UTC]","city":"New York","country":"US","end":"2019-10-30T22:00:00Z[UTC]","visaRequired":false},"hotel":{"address":{"city":"New York","country":"US","street":"street","zipCode":"12345"},"bookingNumber":"XX-012345","name":"Perfect hotel","phone":"09876543"},"traveller":{"address":{"city":"Berlin","country":"Germany","street":"Bakers","zipCode":"100200"},"email":"cristiano@redhat.com","firstName":"Cristiano","lastName":"Nicolai","nationality":"German"}}',
    nodes: [],
    milestones: [],
    childProcessInstances: [
      {
        id: 'c54ca5b0-b975-46e2-a9a0-6a86bf7ac21eajabbcc',
        processId: 'travels',
        businessKey: 'TP444',
        parentProcessInstanceId: '538f9feb-5a14-4096-b791-2055b38da7c6',
        parentProcessInstance: null,
        rootProcessInstanceId: '538f9feb-5a14-4096-b791-2055b38da7c6',
        processName: 'FlightBooking test 2',
        roles: [],
        state: ProcessInstanceState.Active,
        serviceUrl: 'http://localhost:4000',
        lastUpdate: new Date('2019-10-22T03:40:44.089Z'),
        endpoint: 'http://localhost:4000',
        addons: ['process-management'],
        error: {
          nodeDefinitionId: 'a1e139d5-81c77-48c9-84ae-34578e90433n',
          message: 'Something went wrong'
        },
        start: new Date('2019-10-22T03:40:44.089Z'),
        end: new Date('2019-10-22T03:40:44.089Z'),
        variables:
          '{"flight":{"arrival":"2019-10-30T22:00:00Z[UTC]","departure":"2019-10-22T22:00:00Z[UTC]","flightNumber":"MX555"},"trip":{"begin":"2019-10-22T22:00:00Z[UTC]","city":"Berlin","country":"Germany","end":"2019-10-30T22:00:00Z[UTC]","visaRequired":false},"traveller":{"address":{"city":"Karkow","country":"Poland","street":"palna","zipCode":"200300"},"email":"rob@redhat.com","firstName":"Rob","lastName":"Rob","nationality":"Polish"}}',
        nodes: [],
        milestones: [],
        childProcessInstances: []
      }
    ]
  },
  {
    id: '538f9feb-5a14-4096-b791-2055b38da7c6',
    processId: 'travels',
    businessKey: 'Tra234',
    parentProcessInstanceId: null,
    parentProcessInstance: null,
    processName: 'travels',
    rootProcessInstanceId: null,
    roles: [],
    state: ProcessInstanceState.Error,
    addons: ['jobs-management', 'prometheus-monitoring', 'process-management'],
    start: new Date('2019-10-22T03:40:44.089Z'),
    error: {
      nodeDefinitionId: '__a1e139d5-4e77-48c9-84ae-34578e9817n',
      message: 'Something went wrong'
    },
    lastUpdate: new Date('2019-10-22T03:40:44.089Z'),
    serviceUrl: 'http://localhost:4000',
    endpoint: 'http://localhost:4000',
    variables:
      '{"flight":{"arrival":"2019-10-30T22:00:00Z[UTC]","departure":"2019-10-23T22:00:00Z[UTC]","flightNumber":"MX555"},"trip":{"begin":"2019-10-23T22:00:00Z[UTC]","city":"New York","country":"US","end":"2019-10-30T22:00:00Z[UTC]","visaRequired":false},"hotel":{"address":{"city":"New York","country":"US","street":"street","zipCode":"12345"},"bookingNumber":"XX-012345","name":"Perfect hotel","phone":"09876543"},"traveller":{"address":{"city":"Berlin","country":"Germany","street":"Bakers","zipCode":"100200"},"email":"cristiano@redhat.com","firstName":"Cristiano","lastName":"Nicolai","nationality":"German"}}',
    nodes: [],
    milestones: [],
    childProcessInstances: [
      {
        id: 'c54ca5b0-b975-46e2-a9a0-6a86bf7ac21eajabbcc',
        processId: 'travels',
        businessKey: 'TP444',
        parentProcessInstanceId: '538f9feb-5a14-4096-b791-2055b38da7c6',
        parentProcessInstance: null,
        rootProcessInstanceId: '538f9feb-5a14-4096-b791-2055b38da7c6',
        processName: 'FlightBooking test 2',
        roles: [],
        state: ProcessInstanceState.Active,
        serviceUrl: 'http://localhost:4000',
        lastUpdate: new Date('2019-10-22T03:40:44.089Z'),
        endpoint: 'http://localhost:4000',
        addons: ['process-management'],
        error: {
          nodeDefinitionId: 'a1e139d5-81c77-48c9-84ae-34578e90433n',
          message: 'Something went wrong'
        },
        start: new Date('2019-10-22T03:40:44.089Z'),
        end: new Date('2019-10-22T03:40:44.089Z'),
        variables:
          '{"flight":{"arrival":"2019-10-30T22:00:00Z[UTC]","departure":"2019-10-22T22:00:00Z[UTC]","flightNumber":"MX555"},"trip":{"begin":"2019-10-22T22:00:00Z[UTC]","city":"Berlin","country":"Germany","end":"2019-10-30T22:00:00Z[UTC]","visaRequired":false},"traveller":{"address":{"city":"Karkow","country":"Poland","street":"palna","zipCode":"200300"},"email":"rob@redhat.com","firstName":"Rob","lastName":"Rob","nationality":"Polish"}}',
        nodes: [],
        milestones: [],
        childProcessInstances: []
      }
    ]
  }
];
