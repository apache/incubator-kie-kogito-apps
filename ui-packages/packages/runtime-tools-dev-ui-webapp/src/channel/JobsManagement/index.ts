export { JobsManagementGatewayApi } from './JobsManagementGatewayApi';
export * from './JobsManagementContext';
