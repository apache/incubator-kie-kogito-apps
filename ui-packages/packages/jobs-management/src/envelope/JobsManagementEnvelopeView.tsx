import * as React from 'react';
import { useImperativeHandle, useState } from 'react';
import { MessageBusClientApi } from '@kie-tools-core/envelope-bus/dist/api';
import { JobsManagementChannelApi } from '../api';
import JobsManagement from './components/JobsManagement/JobsManagement';
import JobsManagementEnvelopeViewDriver from './JobsManagementEnvelopeViewDriver';
import '@patternfly/patternfly/patternfly.css';

export interface JobsManagementEnvelopeViewApi {
  initialize: () => void;
}
interface Props {
  channelApi: MessageBusClientApi<JobsManagementChannelApi>;
}

export const JobsManagementEnvelopeView = React.forwardRef<
  JobsManagementEnvelopeViewApi,
  Props
>((props, forwardedRef) => {
  const [isEnvelopeConnectedToChannel, setEnvelopeConnectedToChannel] =
    useState<boolean>(false);
  useImperativeHandle(
    forwardedRef,
    () => ({
      initialize: () => {
        setEnvelopeConnectedToChannel(true);
      }
    }),
    []
  );

  return (
    <React.Fragment>
      <JobsManagement
        isEnvelopeConnectedToChannel={isEnvelopeConnectedToChannel}
        driver={new JobsManagementEnvelopeViewDriver(props.channelApi)}
      />
    </React.Fragment>
  );
});

export default JobsManagementEnvelopeView;
