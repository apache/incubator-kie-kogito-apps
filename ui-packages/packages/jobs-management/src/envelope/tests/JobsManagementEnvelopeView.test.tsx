/*
 * Copyright 2021 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from 'react';
import { act } from 'react-dom/test-utils';
import { getWrapper } from '@kogito-apps/components-common';
import { MockedMessageBusClientApi } from './mocks/Mocks';
import JobsManagementEnvelopeView, {
  JobsManagementEnvelopeViewApi
} from '../JobsManagementEnvelopeView';
import JobsManagement from '../components/JobsManagement/JobsManagement';

jest.mock('../components/JobsManagement/JobsManagement');

describe('JobsManagementEnvelopeView tests', () => {
  it('Snapshot', () => {
    const channelApi = new MockedMessageBusClientApi();

    const forwardRef = React.createRef<JobsManagementEnvelopeViewApi>();

    let wrapper = getWrapper(
      <JobsManagementEnvelopeView channelApi={channelApi} ref={forwardRef} />,
      'JobsManagementEnvelopeView'
    );

    expect(wrapper).toMatchSnapshot();

    act(() => {
      if (forwardRef.current) {
        forwardRef.current.initialize();
      }
    });

    wrapper = wrapper.update().find(JobsManagementEnvelopeView);

    expect(wrapper).toMatchSnapshot();

    const inbox = wrapper.find(JobsManagement);

    expect(inbox.exists()).toBeTruthy();
    expect(inbox.props().isEnvelopeConnectedToChannel).toBeTruthy();
    expect(inbox.props().driver).not.toBeNull();
  });
});
