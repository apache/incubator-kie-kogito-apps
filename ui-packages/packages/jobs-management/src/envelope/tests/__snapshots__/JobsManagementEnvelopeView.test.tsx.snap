// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobsManagementEnvelopeView tests Snapshot 1`] = `null`;

exports[`JobsManagementEnvelopeView tests Snapshot 2`] = `
<ForwardRef
  channelApi={
    Object {
      "notifications": Object {
        "jobList__sortBy": [MockFunction],
      },
      "requests": Object {
        "jobList__applyFilter": [MockFunction],
        "jobList__bulkCancel": [MockFunction],
        "jobList__initialLoad": [MockFunction],
        "jobList__query": [MockFunction],
        "jobList_cancelJob": [MockFunction],
        "jobList_rescheduleJob": [MockFunction],
        "jobList_sortBy": [MockFunction],
      },
      "subscribe": [MockFunction],
      "unsubscribe": [MockFunction],
    }
  }
>
  <MockedJobsManagement
    driver={
      JobsManagementEnvelopeViewDriver {
        "channelApi": Object {
          "notifications": Object {
            "jobList__sortBy": [MockFunction],
          },
          "requests": Object {
            "jobList__applyFilter": [MockFunction],
            "jobList__bulkCancel": [MockFunction],
            "jobList__initialLoad": [MockFunction],
            "jobList__query": [MockFunction],
            "jobList_cancelJob": [MockFunction],
            "jobList_rescheduleJob": [MockFunction],
            "jobList_sortBy": [MockFunction],
          },
          "subscribe": [MockFunction],
          "unsubscribe": [MockFunction],
        },
      }
    }
    isEnvelopeConnectedToChannel={true}
  />
</ForwardRef>
`;
